### PR TITLE
Don't overwrite `self.teams` upon parsing responders in OpsGenie integration

### PR DESCRIPTION
**Pull request by lstyles**
## Checklist

- [x] I have reviewed the [contributing guidelines](https://github.com/jertel/elastalert2/blob/master/CONTRIBUTING.md).
- [x] I have included unit tests for my changes or additions.
- [x] I have successfully run `make test-docker` with my changes.
- [x] I have manually tested all relevant modes of the change in this PR.
- [x] I have updated the [documentation](https://elastalert2.readthedocs.io).
- [x] I have updated the [changelog](https://github.com/jertel/elastalert2/blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by jertel** [comment]
Thanks for submitting this. There are already extensive unit tests for this alerter, so adjusting one of them to send two alerts and then assert that the second one was unaffected by the first should (hopefully) be low-effort.

**Comment by lstyles** [comment]
Thanks. Updated, but the changes needed were a bit more extensive than I would have liked.

I've added a test for this particular issue, but had to make some other changes to tidy things up a bit. Some of the tests were checking for (what I think) were incorrect values and some others were checking properties on the alert object instead of mocked post request.

Let me know what you think. Apologies if it is a mess, I've not done any Python dev prior to this.

File: elastalert/alerters/opsgenie.py [hunk 1]
```diff
@@ -35,25 +35,29 @@ def __init__(self, *args):
...
-                if not default_responders:
-                    elastalert_logger.warning("OpsGenieAlerter: default responder not set. Falling back")
-                    formated_responders = responders
-                else:
-                    formated_responders = default_responders
-            responders = formated_responders
+        if responders is None:
+            return None
+        if responder_args is None:
+            responder_args = dict()
```
Comment (lstyles) [inline elastalert/alerters/opsgenie.py:41]: The top two lines are potentially redundant.

The bottom two lines fix a scenario where `responders` contains a placeholder, but `responders_args` isn't configured in a rule at all. Previously, a text with placeholder would end up as a rule. Now, a warning will be shown and `default_responders` will be used instead.

**Comment by lstyles** [comment]
The two new tests could probably be named to better describe what they're doing:
- preserve_rule_config_upon_parsing_teams
- preserve_rule_config_upon_parsing_recipients
or something like that

**Comment by nsano-rururu** [comment]
The next version, 2.21.0, will allow both the old setting name and the correct wording. However, if you use the old setting name, how about informing the user that the old setting value will be deleted when the logger reaches 2.25.0 every time an alert notification is sent?

**Comment by nsano-rururu** [comment]
For warning loggers, use elastalert_logger.warning.

example
https://github.com/jertel/elastalert2/pull/793/files

**Comment by lstyles** [comment]
I've put something together but on alerter initialisation instead of alert send. Deprecation logic feels more in place there vs having it in the alert method.

**Comment by nsano-rururu** [comment]
Update CHANGELOG.md.

**Comment by nsano-rururu** [comment]
@jertel 

Any other comments?

**Review by jertel — APPROVED** [review]

**Comment by jertel** [comment]
Looks good, thank you both. The schema.yaml is missing those two properties, as well as other OpsGenie properties. But that can wait for another PR since it's not directly related to this.

**Comment by lstyles** [comment]
https://github.com/jertel/elastalert2/pull/1543 schema update as a separate PR (1 - to not block this one, 2 - because I'm not 100% sure it's correct)

**Pull request merged**